Unable to compile with libmdns-0.9.2: arguments to this method are incorrect

### Description
Build fails recently in Docker container `rust:slim-bookworm`.

I am not sure what exactly changed recently but I have been using the same Docker container to build librespot for a while now and it recently stopped working.

I don't know enough about Rust to debug myself, but I can provide any addition information that is required.

### Version
I tried the latest `0.7.1` as well as `0.7.0` (which worked previously).

### How to reproduce
Use the following `Dockerfile` to reproduce:

```Dockerfile
FROM rust:slim-bookworm AS builder
ARG LIBRESPOT_VERSION=0.7.1
 
RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
    apt-get full-upgrade -y && \
    apt-get install -y libasound2-dev pkg-config libssl-dev

RUN cargo install librespot --version "${LIBRESPOT_VERSION}"
```

### Log

```
51.41 error[E0308]: arguments to this method are incorrect
51.41    --> /usr/local/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/librespot-discovery-0.7.1/src/lib.rs:409:33
51.41     |
51.41 409 |             let svc = responder.register(
51.41     |                                 ^^^^^^^^
51.41 410 |                 DNS_SD_SERVICE_NAME.to_owned(),
51.41     |                 ------------------------------ expected `&str`, found `String`
51.41 411 |                 name.into_owned(),
51.41     |                 ----------------- expected `&str`, found `String`
51.41     |
51.41 note: method defined here
51.41    --> /usr/local/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/libmdns-0.9.2/src/lib.rs:292:12
51.41     |
51.41 292 |     pub fn register(&self, svc_type: &str, svc_name: &str, port: u16, txt: &[&str]) -> Service {
51.41     |            ^^^^^^^^
51.41 help: try removing the method call
51.41     |
51.41 410 -                 DNS_SD_SERVICE_NAME.to_owned(),
51.41 410 +                 DNS_SD_SERVICE_NAME,
51.41     |
51.41 help: consider borrowing here
51.41     |
51.41 411 |                 &name.into_owned(),
51.41     |                 +
51.41 
51.51 For more information about this error, try `rustc --explain E0308`.
51.51 error: could not compile `librespot-discovery` (lib) due to 1 previous error
51.51 warning: build failed, waiting for other jobs to finish...
60.07 error: failed to compile `librespot v0.7.0`, intermediate artifacts can be found at `/tmp/cargo-installA5cVQf`.
60.07 To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

### Host (what you are running `librespot` on):
- OS: Arch Linux with Docker (build and running librespot + snapcast inside Docker)
- Platform: x86_64

--- a/discovery/Cargo.toml
+++ b/discovery/Cargo.toml
@@ -41,7 +41,7 @@ hyper-util = { version = "0.1", features = [
     "server-graceful",
     "service",
 ] }
-libmdns = { version = "0.9", optional = true }
+libmdns = { version = "0.10", optional = true }
 log = "0.4"
 rand = "0.9"
 serde = { version = "1", default-features = false, features = [

--- a/discovery/src/lib.rs
+++ b/discovery/src/lib.rs
@@ -406,12 +406,7 @@ fn launch_libmdns(
             }
             .map_err(|e| DiscoveryError::DnsSdError(Box::new(e)))?;
 
-            let svc = responder.register(
-                DNS_SD_SERVICE_NAME.to_owned(),
-                name.into_owned(),
-                port,
-                &TXT_RECORD,
-            );
+            let svc = responder.register(&DNS_SD_SERVICE_NAME, &name, port, &TXT_RECORD);
 
             let _ = shutdown_rx.blocking_recv();
 

